docs/guides/unity-catalog.md

```diff
@@ -0,0 +1,530 @@
+  provider          = databricks.workspace
+  external_location = databricks_external_location.some.id
+  grant {
+    principal  = "Data Engineers"
+    privileges = ["CREATE TABLE", "READ FILES"]
+  }
+}
+```
+
+## Configure Unity Catalog clusters
+
+To ensure the integrity of ACLs, users are required to access Unity Catalog through compute resources configured with strong isolation guarantees and other security features. This is achieved through a new concept called ‘Security Modes’, which are applied to compute resources.
+
+To use Unity Catalog from a regular [databricks_cluster](../resources/cluster.md), one must configure the cluster with a proper **Data Security Mode**.  Currently, Unity Catalog is supported in two security modes: **User Isolation** and **Single User**.
+
+- **User Isolation** clusters can be shared by multiple users, but only SQL language is allowed. Some advanced cluster features such as library installation, init scripts and the DBFS Fuse mount are also disabled in this mode to ensure security isolation among cluster users.
+
+```hcl
+data "databricks_spark_version" "latest" {
+  provider = databricks.workspace
+}
+data "databricks_node_type" "smallest" {
+  provider   = databricks.workspace
+  local_disk = true
+}
+
+resource "databricks_cluster" "unity_sql" {
+  provider                = databricks.workspace
+  cluster_name            = "Unity SQL"
+  spark_version           = data.databricks_spark_version.latest.id
+  node_type_id            = data.databricks_node_type.smallest.id
+  autotermination_minutes = 60
+  enable_elastic_disk     = false
+  num_workers             = 2
+  aws_attributes {
+    availability = "SPOT"
+  }
+  data_security_mode = "USER_ISOLATION"
+}
+```
+
+- To use those advanced cluster features or languages like Python, Scala and R with Unity Catalog, one must choose **Single User** Mode when launching the cluster. The cluster can only be used exclusively by a single user (by default the owner of the cluster); other users are not allowed to attach to the cluster.
+This means a group of users, which is managed as a group through SCIM provisioning, will a collection of single-user [databricks_cluster](../resources/cluster.md), which they should be able to restart. Terraform's `for_each` meta-attribute helps to do this easily.
+
+First we use [databricks_group](../data-sources/group.md) and [databricks_user](../data-sources/user.md) data resources to get the list of user names, that belong to a group
+
+```hcl
+data "databricks_group" "dev" {
+  provider     = databricks.workspace
+  display_name = "dev-clusters"
+}
+
+data "databricks_user" "dev" {
+  provider = databricks.workspace
+  for_each = data.databricks_group.dev.members
+  user_id  = each.key
+}
+```
+
+Once we have a specific list of user resources, we could proceed creating single-user clusters and provide permissions with `for_each = data.databricks_user.dev` to ensure it's done for each user:
```

Comment:
Why are you preferring single-user clusters for this walkthrough? They have drawbacks when it comes to views.

Reply:
Which drawbacks, by the way?..

Reply:
I think @mdlinville is referring to the current limitation: `Fine grained access controls via dynamic views are not available on clusters in ‘Single User’`

We are creating both user isolation & single user clusters in this guide, user isolation for shared sql and single-user for Python, Scala, R and MLR

Reply:
@nkvuong can you maybe add couple of sentences to elaborate on this? :)